### PR TITLE
ARTEMIS-3043 - improvements on new console

### PR DESCRIPTION
**Pull request by andytaylor**
https://issues.apache.org/jira/browse/ARTEMIS-3043

### CODE REVIEW

**Comment by sebthom** [comment]
@andytaylor do you think you could also fix server side sorting? I realized yesterday that for example on the addresses and queues view the sorting does not work across across pages. I think it is because column ids in the Javascript files are upper case but the server control mbean uses lower or camelcase amd thus does not interpret the selected sort column correctly.

File: artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/addresses.js [hunk 1]
```diff
@@ -59,17 +60,43 @@ var Artemis;
...
+
         ctrl.filter = {
             fieldOptions: [
-                {id: 'ID', name: 'ID'},
-                {id: 'NAME', name: 'Name'},
-                {id: 'ROUTING_TYPES', name: 'Queue Count'},
-                {id: 'QUEUE_COUNT', name: 'User'}
+                {id: 'id', name: 'ID'},
+                {id: 'name', name: 'Name'},
+                {id: 'routingTypes', name: 'Queue Count'},
```
Comment (sebthom) [inline artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/addresses.js:98]: Can you also fix the name attribute for this and the next field? I think they are mixed up.

**Comment by sebthom** [comment]
Never mind, it is already fixed by your PR. It just isn't mentioned in the JIRA issue.

File: artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js [hunk 2]
```diff
@@ -59,28 +60,76 @@ var Artemis;
...
+        ctrl.pagination.reset();
         var mbean = Artemis.getBrokerMBean(workspace, jolokia);
         ctrl.allAddresses = [];
         ctrl.queues = [];
         ctrl.workspace = workspace;
         ctrl.refreshed = false;
+        ctrl.dtOptions = {
+           // turn of ordering as we do it ourselves
+           ordering: false,
+           columns: [
```
Comment (michaelandrepearce) [inline artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js:72]: Queue Enabled/Disabled field is missing

File: artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js [hunk 3]
```diff
@@ -59,28 +60,76 @@ var Artemis;
...
+                  {name: "Temporary", visible: false},
+                  {name: "Auto Created", visible: false},
+                  {name: "User", visible: false},
+                  {name: "Total Messages Added", visible: false},
+                  {name: "Total Messages Acked", visible: false},
+                  {name: "Delivering Count", visible: false},
+                  {name: "Messages Killed", visible: false},
+                  {name: "Direct Deliver", visible: false},
+                  {name: "Exclusive", visible: false},
+                  {name: "last value", visible: false},
```
Comment (michaelandrepearce) [inline artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js:94]: **L**ast**V**alue

Comment (michaelandrepearce) [inline artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js:94]: What about LastValueKey

File: artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js (hunk 2 above)
Comment (michaelandrepearce) [inline artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js:72]: Ring Size field is missing

File: artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js [hunk 4]
```diff
@@ -59,28 +60,76 @@ var Artemis;
...
+           columns: [
+                  {name: "ID", visible: true},
+                  {name: "name", visible: true},
+                  {name: "Address", visible: true},
+                  {name: "Routing Type", visible: true},
+                  {name: "Filter", visible: true},
+                  {name: "Durable", visible: true},
+                  {name: "Max Consumers", visible: true},
+                  {name: "Purge On No Consumers", visible: true},
+                  {name: "Consumer Count", visible: true},
```
Comment (michaelandrepearce) [inline artemis-hawtio/artemis-plugin/src/main/webapp/plugin/js/components/queues.js:81]: a few more queue fields/values missing: "Consumers Before Dispatch" & "Delay Before Dispatch"

**Comment by michaelandrepearce** [comment]
Can we add back ability to change column widths?

**Comment by andytaylor** [comment]
> Can we add back ability to change column widths?

Unfortunately I have no idea whether this is even possible, there is nothing in datatables to do this which i what hawtIO2 now uses.

**Comment by andytaylor** [comment]
updated PR after feedback. fyi some of the table columns weren't even exposed in the server side views, in the new year I'm going to add some tests to catch these being missed when attributes are added to queues etc

**Comment by andytaylor** [comment]
just fyi, I'll try and address column resizing separately, see if i can find a solution with datatables and dt-options

**Comment by michaelandrepearce** [comment]
I’m not 100% which datatables impl you’re using but a quick Google fu:

https://datatables.net/forums/discussion/26786/basic-column-resizing-plugin

https://www.gyrocode.com/articles/jquery-datatables-column-reordering-and-resizing/

**Comment by andytaylor** [comment]
cheers @michaelandrepearce i'll take a look in  the new year

**Comment by michaelandrepearce** [comment]
Why has this closed?

**Comment by clebertsuconic** [comment]
@michaelandrepearce it's merged.. shouldn't it be merged?

**Comment by michaelandrepearce** [comment]
It was waiting on Andy to look at column resizing which he was going to sort this side of new year and was waiting on that.

**Comment by michaelandrepearce** [comment]
Also i raised this on the original PR's that did the hawtio2 change, but seems the comments i left got missed with the changes to console being done, can the docs please be updated.

**Comment by andytaylor** [comment]
Column resizing might not even be possible, Im investigating now but either way it wont be available in the short term if at all. I would merge this as is I dont think we need to wait for resizing.

**Comment by andytaylor** [comment]
@michaelandrepearce I'll update the docs before next week

**Comment by andytaylor** [comment]
just fyi my column resizing wip is at https://github.com/andytaylor/activemq-artemis/tree/resizing

im currently blocked on at first glance looks like a bug in Angular

**Pull request closed without merging**